src/core/radio/ble_secure.hpp

```diff
@@ -0,0 +1,521 @@
+     * including freeing @p aMessage if the message buffer is no longer needed.
+     *
+     * @retval kErrorNone          Successfully sent message.
+     * @retval kErrorNoBufs        Failed to allocate buffer memory.
+     * @retval kErrorInvalidState  TLS connection was not initialized.
+     *
+     */
+    Error SendMessage(Message &aMessage);
+
+    /**
+     * Sends a secure BLE data packet.
+     *
+     * @param[in]  aBuf            A pointer to the data to send.
+     * @param[in]  aLength         A number indicating the length of the data buffer.
+     *
+     * @retval kErrorNone          Successfully sent data.
+     * @retval kErrorNoBufs        Failed to allocate buffer memory.
+     * @retval kErrorInvalidState  TLS connection was not initialized.
+     *
+     */
+    Error Send(uint8_t *aBuf, uint16_t aLength);
+
+    /**
+     * Sends a secure BLE data packet containing a TCAT application TLV.
+     *
+     * @param[in]  aBuf            A pointer to the data to send.
+     * @param[in]  aLength         A number indicating the length of the data buffer.
+     *
+     * @retval kErrorNone          Successfully sent data.
+     * @retval kErrorNoBufs        Failed to allocate buffer memory.
+     * @retval kErrorInvalidState  TLS connection was not initialized.
+     *
+     */
+    Error SendApplicationTlv(uint8_t *aBuf, uint16_t aLength);
+
+    /**
+     * Sends all remaining bytes in the send buffer.
+     *
+     * @retval kErrorNone          Successfully enqueued data into the output interface.
+     * @retval kErrorNoBufs        Failed to allocate buffer memory.
+     * @retval kErrorInvalidState  TLS connection was not initialized.
+     *
+     */
+    Error Flush(void);
+
+    /**
+     * Used to pass data received over a BLE link to the secure BLE server.
+     *
+     * @param[in]  aBuf            A pointer to the data received.
+     * @param[in]  aLength         A number indicating the length of the data buffer.
+     *
+     */
+    Error HandleBleReceive(uint8_t *aBuf, uint16_t aLength);
+
+    /**
+     * Used to notify the secure BLE server that a BLE Device has been connected.
+     *
+     * @param[in]  aConnectionId    The identifier of the open connection.
+     *
+     */
+    void HandleBleConnected(uint16_t aConnectionId);
+
+    /**
+     * Used to notify the secure BLE server that the BLE Device has been disconnected.
+     *
+     * @param[in]  aConnectionId    The identifier of the open connection.
+     *
+     */
+    void HandleBleDisconnected(uint16_t aConnectionId);
+
+    /**
+     * Used to notify the secure BLE server that the BLE Device has updated ATT_MTU size.
+     *
+     * @param[in]  aMtu             The updated ATT_MTU value.
+     *
+     */
+    Error HandleBleMtuUpdate(uint16_t aMtu);
+
+private:
+    /**
+     * BLE secure state.
+     *
+     */
+    enum BleState : uint8_t
```

Comment:
Can remove docs/comment for `private` definition?
```suggestion
    enum BleState : uint8_t
```

Reply:
fixed.

Reply:
fixed.